src/libraries/System.Net.Security/src/System/Net/Security/SslStream.IO.cs

```diff
@@ -54,10 +54,10 @@ private void CloseInternal()
 
             // Ensure a Read or Auth operation is not in progress,
             // block potential future read and auth operations since SslStream is disposing.
-            // This leaves the _nestedRead = 1 and _nestedAuth = 1, but that's ok, since
+            // This leaves the _nestedRead = 2 and _nestedAuth = 2, but that's ok, since
             // subsequent operations check the _exception sentinel first
-            if (Interlocked.Exchange(ref _nestedRead, 1) == 0 &&
-                Interlocked.Exchange(ref _nestedAuth, 1) == 0)
+            if (Interlocked.Exchange(ref _nestedRead, (int)StreamUse.Disposed) == (int)StreamUse.NotInUse &&
```

Comment:
`Interlocked.Exchange` has an overload which would work for enums, right? I am bothered by the number of casts this requires.

Reply:
> overload which would work for enums

Not yet: #65184

Reply:
I see, then we might want to consider making these values int constants

Reply:
I was thinking about it but at the end decided to do the enum. It would be easy to fix if we prefer constant to avoid casting.

Reply:
updated.